packages/runtime/client/rest/src/contract/details.ts

```diff
@@ -43,25 +54,42 @@ export const Payout = t.type({
  * @interface
  * @category Endpoint : Get Contract By Id
  */
-export interface ContractDetails extends t.TypeOf<typeof ContractDetails> {}
+export type ContractDetails = {
+  version: MarloweVersion;
+  contractId: ContractId;
+  initialContract: Contract;
+  roleTokenMintingPolicyId: PolicyId;
+  status: TxStatus;
+  tags: Tags;
+  metadata: Metadata;
+  unclaimedPayouts: Payout[];
+} /** only available when the contract is Active */
+& { currentContract?: Contract; state?: MarloweState ,utxo?: TxOutRef } 
+/** only available when the contract is confirmed */
+& {block?: BlockHeader;} 
+& {txBody?: TextEnvelope } 
+
```

Comment:
Active >> confirmed, rigth?
If that is true, then block and txBody are available when confirmed and when active? or just with confirmed?

Reply:
Active is "confirmed" except in a Closed state

Reply:
```haskell
instance ToDTOWithTxStatus (Tx.ContractCreated v) where
  toDTOWithTxStatus status (Tx.ContractCreated era Tx.ContractCreatedInEra{..}) =
    Web.ContractState
      { contractId = toDTO contractId
      , roleTokenMintingPolicyId = toDTO rolesCurrency
      , version = case version of
          MarloweV1 -> Web.V1
      , tags = foldMap fst $ toDTO $ marloweMetadata metadata
      , continuations = snd =<< toDTO (marloweMetadata metadata)
      , metadata = toDTO $ transactionMetadata metadata
      , status = toDTO status
      , block = Nothing
      , initialContract = case version of
          MarloweV1 -> Sem.marloweContract datum
      , currentContract = case version of
          MarloweV1 -> Just $ Sem.marloweContract datum
      , state = case version of
          MarloweV1 -> Just $ Sem.marloweState datum
      , assets = toDTO assets
      , utxo = Nothing
      , txBody = case status of
          Unsigned -> Just case era of
            ReferenceTxInsScriptsInlineDatumsInBabbageEra -> toDTO txBody
            ReferenceTxInsScriptsInlineDatumsInConwayEra -> toDTO txBody
          Submitted -> Nothing
      , unclaimedPayouts = []
      }
instance ToDTO SomeContractState where
  toDTO (SomeContractState MarloweV1 ContractState{..}) =
    Web.ContractState
      { contractId = toDTO contractId
      , roleTokenMintingPolicyId = toDTO roleTokenMintingPolicyId
      , version = Web.V1
      , tags = foldMap fst $ toDTO $ marloweMetadata metadata
      , continuations = snd =<< toDTO (marloweMetadata metadata)
      , metadata = toDTO $ transactionMetadata metadata
      , status = Web.Confirmed
      , block = Just $ toDTO initialBlock
      , initialContract = Sem.marloweContract $ datum initialOutput
      , currentContract = Sem.marloweContract . datum <$> latestOutput
      , state = Sem.marloweState . datum <$> latestOutput
      , assets = maybe emptyAssets (\Core.TransactionScriptOutput{..} -> toDTO assets) latestOutput
      , utxo = toDTO . utxo <$> latestOutput
      , txBody = Nothing
      , unclaimedPayouts =
          (\(payoutId, Core.Api.Payout{..}) -> Web.Payout (toDTO payoutId) (toDTO . tokenName $ datum) (toDTO assets))
            <$> M.toList unclaimedPayouts
      }

data ContractState v = ContractState
  { contractId :: ContractId
  , roleTokenMintingPolicyId :: PolicyId
  , metadata :: MarloweTransactionMetadata
  , initialBlock :: BlockHeader
  , initialOutput :: TransactionScriptOutput v
  , latestBlock :: BlockHeader
  , latestOutput :: Maybe (TransactionScriptOutput v)
  , unclaimedPayouts :: Map TxOutRef (Payout v)
  }
  deriving (Generic)


```

Reply:
```json
/** Only available when the contract is Unsigned/Submitted and Active */
& { currentContract?: Contract; state?: MarloweState }
  /** Only available when Active */
& {utxo?: TxOutRef } 
  /** Only available when the contract is confirmed */
& {block?: BlockHeader;} 
  /** Only avaible When the contract is Unsigned/Submitted */
& {txBody?: TextEnvelope } 
```

Reply:
I have updated it